### PR TITLE
fix JsonNode deserialising from null

### LINKED ISSUE #85172: Deserialising null to JsonNode in nested structures fails in System.Text.Json
`JsonSerializer.Deserialize<JsonNode>("null")` returns null
`JsonSerializer.Deserialize<List<JsonNode>>("[null]")` throws
`JsonSerializer.Deserialize<Dictionary<string, JsonNode>>("{ \"Foo\": null }")` throws

Assuming this isn't intentional? Reproducible in .Net 6 & 7.

### PR DESCRIPTION
**Pull request by adj123**
Fixes #85172

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonNodeTests.cs [hunk 1]
```diff
@@ -39,6 +41,10 @@ public static void JsonTypes_Deserialize()
             Assert.IsAssignableFrom<JsonValue>(JsonNode.Parse("\"str\""));
             Assert.IsAssignableFrom<JsonValue>(JsonNode.Parse(ToUtf8("\"str\"")));
             Assert.IsType<JsonElement>(JsonSerializer.Deserialize<object>("\"str\""));
+
+            Assert.Null(JsonSerializer.Deserialize<JsonNode>("null"));
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonNodeTests.cs:45]: Would it be possible to 

1. Also test deserializing an object containing a `JsonNode` property?
2. Add tests validating the same behavior for `JsonObject` and `JsonArray`?

Comment (adj123) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonNodeTests.cs:45]: Sure thing!

**Review by eiriktsarpalis — APPROVED** [review]
Thanks!

**Comment by eiriktsarpalis** [comment]
@adj123 would you mind responding to the CLA bot as well?

https://github.com/dotnet/runtime/pull/85463#issuecomment-1525696202

Thanks!

**Comment by adj123** [comment]
@dotnet-policy-service agree [company="DRW Holdings"]

**Pull request merged**